Clad picks up system LLVM

Even if `-DCLAD_PATH_TO_LLVM_BUILD` is passed, the system llvm is discovered first.

```
 /cvmfs/sft.cern.ch/lcg/releases/CMake/3.11.1-773ff/x86_64-centos7-gcc62-opt/bin/cmake  -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=/cvmfs/projects.cern.ch/intelsw/psxe/linux/x86_64/2018/compilers_and_libraries_2018.2.199/linux/bin/intel64/icc -DCMAKE_C_FLAGS= -restrict -wd1572 -m64 -wd279 -wd873 -wd2536 -wd597 -wd1098 -wd1292 -wd1478 -wd3373 -pthread -fPIC -Werror=date-time -w -ffunction-sections -fdata-sections -DCMAKE_CXX_COMPILER=/cvmfs/projects.cern.ch/intelsw/psxe/linux/x86_64/2018/compilers_and_libraries_2018.2.199/linux/bin/intel64/icc -DCMAKE_CXX_FLAGS= -wd1476 -wd1572 -m64 -wd279 -wd873 -wd2536 -wd597 -wd1098 -wd1292 -wd1478 -wd3373 -pthread -std=c++11  -fPIC -fvisibility-inlines-hidden -Werror=date-time -std=c++11 -w -ffunction-sections -fdata-sections -fno-common -Woverloaded-virtual -fno-strict-aliasing -Wno-nested-anon-types -Wno-covered-switch-default -Wno-unused-local-typedef -fno-rtti -DCMAKE_INSTALL_PREFIX=/data/sftnight/workspace/root-benchmark/BUILDTYPE/Release/COMPILER/icc18/LABEL/performance-sandy-cc7/build/etc/cling/plugins/ -DCLAD_PATH_TO_LLVM_BUILD=/data/sftnight/workspace/root-benchmark/BUILDTYPE/Release/COMPILER/icc18/LABEL/performance-sandy-cc7/build/interpreter/llvm/src -G"Unix Makefiles" /data/sftnight/workspace/root-benchmark/BUILDTYPE/Release/COMPILER/icc18/LABEL/performance-sandy-cc7/build/interpreter/llvm/src/tools/cling/tools/plugins/clad/clad-prefix/src/clad
-- Could NOT find Subversion (missing: Subversion_SVN_EXECUTABLE) 
CMake Error at /opt/llvm-5.0.1/lib64/cmake/llvm/LLVMExports.cmake:960 (message):
  The imported target "LLVMDemangle" references the file

     "/opt/llvm-5.0.1/lib64/libLLVMDemangle.a"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "/opt/llvm-5.0.1/lib64/cmake/llvm/LLVMExports.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  /opt/llvm-5.0.1/lib64/cmake/llvm/LLVMConfig.cmake:188 (include)
  CMakeLists.txt:8 (find_package)


-- Configuring incomplete, errors occurred!
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,7 +5,7 @@ if( CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR )
   cmake_minimum_required(VERSION 3.4.3)
   find_package(Subversion)
 
-  find_package(LLVM 5.0 REQUIRED CONFIG PATHS ${CLAD_PATH_TO_LLVM_BUILD})
+  find_package(LLVM 5.0 REQUIRED CONFIG PATHS ${CLAD_PATH_TO_LLVM_BUILD} NO_DEFAULT_PATH)
 
   # We cannot use find_package(Clang) or include(ClangConfig) because we require
   # a specific version (currently 5.0) and ClangConfig.cmake tries to do

